WcaOnRails/app/models/team.rb

```diff
@@ -18,4 +18,23 @@ def membership_periods_cannot_overlap_for_single_user
       end
     end
   end
+
+  validate :member_cannot_be_empty
+  def member_cannot_be_empty
+    team_members.each do |member|
+      if member.user == nil
+        errors[:base] << "Team member cannot be empty"
+      end
+    end
+  end
+
+  validate :start_date_cannot_be_empty
+  def start_date_cannot_be_empty
+    team_members.each do |member|
+      if member.start_date == nil
+        errors[:base] << "Membership start cannot be empty"
+      end
+    end
+  end
+
```

Comment:
You iterate over all team members and validate each of them, shouldn't the validation be a part of TeamMember?